ical_support.c: bump refcount of rrules in multirrule iterator

rsto opened the pull request:
Fixes a memory leak that got introduced by libical's recent change to rrule-by-ref.

Run cunit tests with valgrind to reproduce the issue:

```
make VG=1 check-ical_support
```

ksmurchison reviewed (commented):
Do we actually want these #ifdef protected?  Won't we leak recur->rscale in the non-by-ref case?

rsto commented:
I don't have an old-enough libical around to check this. Is the unref function also available without recur-by-ref

ksmurchison commented:
> I don't have an old-enough libical around to check this. Is the unref function also available without recur-by-ref

Yes, I defined one in ical_support.c:3299

ksmurchison commented on imap/ical_support.c:
```diff
@@ -516,7 +516,7 @@ static void multirrule_iterator_add(struct multirrule_iterator *iter,
     entry->next = icalrecur_iterator_next(entry->icaliter);
     entry->recur = recur;
-    entry->recur->rscale = xstrdupnull(recur->rscale);
+    icalrecurrencetype_ref(recur);
```
Hmm.  I misread this initially as icalrecurrencetype_unref().  We don't have a non-recur-by-ref version of this function.  We could make a macro which does nothing or #ifdef protect this.

rsto replied:
alright. I'll re-add the unref then.

rsto replied:
I mean, I'll re-add guarding the unref then.

elliefm commented:
I might have an old enough cyuslibs still, I'll pull this branch and see what happens

elliefm commented:
Is there something I can tell you about my cyruslibs install that will confirm whether it's old enough?  Oh actually (happening as I type), here's a thing -- ical_support no longer building for me:

```
imap/ical_support.c: In function 'multirrule_iterator_add':
imap/ical_support.c:519:5: error: implicit declaration of function 'icalrecurrencetype_ref'; did you mean 'icalrecurrencetype_unref'? [-Werror=implicit-function-declaration]
  519 |     icalrecurrencetype_ref(recur);
      |     ^~~~~~~~~~~~~~~~~~~~~~
      |     icalrecurrencetype_unref
  CC       imap/libcyrus_imap_la-webdav_db.lo
cc1: all warnings being treated as errors
```

ksmurchison commented:
@elliefm Eithwr we need to #ifdef protect that call of create an empty macro for that function

rsto committed:
ical_support.c: bump refcount of rrules in multirrule iterator

Fixes a memory leak that got introduced by libical's recent
change to rrule-by-ref.

Run cunit tests with valgrind to reproduce the issue:

```
make VG=1 check-ical_support
```

rsto commented:
Now compiles and tests/valgrinds cleanly with both recur-by-ref and recur-by-value libical.

ksmurchison reviewed (approved):
If it works and is memory clean, WFM
